packages/components/button/src/button.vue

```diff
@@ -48,120 +48,89 @@ import {
-  },
-
-  props: buttonProps,
-  emits: buttonEmits,
-
-  setup(props, { emit, slots }) {
-    const buttonRef = ref()
-    const buttonGroupContext = inject(buttonGroupContextKey, undefined)
-    const globalConfig = useGlobalConfig('button')
-    const ns = useNamespace('button')
-    const autoInsertSpace = computed(
-      () =>
-        props.autoInsertSpace ?? globalConfig.value?.autoInsertSpace ?? false
-    )
-
-    // add space between two characters in Chinese
-    const shouldAddSpace = computed(() => {
-      const defaultSlot = slots.default?.()
-      if (autoInsertSpace.value && defaultSlot?.length === 1) {
-        const slot = defaultSlot[0]
-        if (slot?.type === Text) {
-          const text = slot.children
-          return /^\p{Unified_Ideograph}{2}$/u.test(text as string)
-        }
-      }
-      return false
-    })
-
-    const { form } = useFormItem()
-    const buttonSize = useSize(computed(() => buttonGroupContext?.size))
-    const buttonDisabled = useDisabled()
-    const buttonType = computed(
-      () => props.type || buttonGroupContext?.type || ''
-    )
-
-    // calculate hover & active color by color
-    const typeColor = computed(
-      () => useCssVar(`--el-color-${props.type}`).value
-    )
-    const buttonStyle = computed(() => {
-      let styles = {}
-
-      const buttonColor = props.color || typeColor.value
-
-      if (buttonColor) {
-        const shadeBgColor = new TinyColor(buttonColor).shade(10).toString()
-        if (props.plain) {
-          styles = {
-            '--el-button-bg-color': new TinyColor(buttonColor)
-              .tint(90)
-              .toString(),
-            '--el-button-text-color': buttonColor,
-            '--el-button-hover-text-color': 'var(--el-color-white)',
-            '--el-button-hover-bg-color': buttonColor,
-            '--el-button-hover-border-color': buttonColor,
-            '--el-button-active-bg-color': shadeBgColor,
-            '--el-button-active-text-color': 'var(--el-color-white)',
-            '--el-button-active-border-color': shadeBgColor,
-          }
-        } else {
-          const tintBgColor = new TinyColor(buttonColor).tint(20).toString()
-          styles = {
-            '--el-button-bg-color': buttonColor,
-            '--el-button-border-color': buttonColor,
-            '--el-button-hover-bg-color': tintBgColor,
-            '--el-button-hover-border-color': tintBgColor,
-            '--el-button-active-bg-color': shadeBgColor,
-            '--el-button-active-border-color': shadeBgColor,
-          }
-        }
+const props = defineProps(buttonProps)
+const emit = defineEmits(buttonEmits)
+const slots = useSlots()
+
+const buttonGroupContext = inject(buttonGroupContextKey, undefined)
+const globalConfig = useGlobalConfig('button')
+const ns = useNamespace('button')
+const { form } = useFormItem()
+const _size = useSize(computed(() => buttonGroupContext?.size))
+const _disabled = useDisabled()
+const _ref = ref()
```

Comment:
Can we add annotation for this one to refer to `HTMLButtonElement`? Which enriches the typing system

Reply:
Fixed.